pkg/broker/logic.go

```diff
@@ -23,20 +23,28 @@ type BusinessLogic interface {
 	// which services they provide. GetCatalog is called when a platform makes
 	// initial contact with the broker to find out about that broker's services.
 	//
+	// The parameters are:
+	// - a RequestContext object which encapsulates:
+	//    - a response writer, in case fine-grained control over the response is
+	//      required
+	//    - the original http request, in case access is required (to get special
+	//      request headers, for example)
+	//
 	// For more information, see:
 	//
 	// https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#catalog-management
-	GetCatalog(w http.ResponseWriter, r *http.Request) (*osb.CatalogResponse, error)
+	GetCatalog(c *RequestContext) (*osb.CatalogResponse, error)
 	// Provision encapsulates the business logic for a provision operation and
 	// returns a osb.ProvisionResponse or an error. Provisioning creates a new
 	// instance of a particular service.
 	//
 	// The parameters are:
 	// - a osb.ProvisionRequest created from the original http request
-	// - a response writer, in case fine-grained control over the response is
-	//   required
-	// - the original http request, in case access is required (to get special
-	//   request headers, for example)
+	// - a RequestContext object which encapsulates:
+	//    - a response writer, in case fine-grained control over the response is
```

Comment:
Let me know if you find these useful, I kept it, but I think it might be better to get rid of it? As they are already documented in the `RequestContext` and if we decide to add something else to that we would need to adjust in each of these comments.

Reply:
I think it's fine for now, especially since I think the right level for the doc on this interface is to fully spell stuff like this out.